lib/Reader/readerir.cpp

```diff
@@ -3675,6 +3746,16 @@ IRNode *GenIR::addressOfValue(IRNode *Leaf) {
     StoreInst *Store = LLVMBuilder->CreateStore(Leaf, Alloc);
     return (IRNode *)Alloc;
   }
+  case Type::TypeID::PointerTyID: {
+    assert(doesValueRepresentStruct(Leaf));
+
+    // Create a new pointer to the struct. The new pointer will point to the
+    // same struct as the original pointer but doesValueRepresentStruct will
+    // return false for the new pointer.
+    Instruction *Alloc = createTemporary(LeafTy);
+    StoreInst *Store = LLVMBuilder->CreateStore(Leaf, Alloc);
+    return (IRNode *)LLVMBuilder->CreateLoad(Alloc);
+  }
```

Comment:
I want to keep an invariant that a pointer value either represents a struct or a pointer to struct. If we don't have that invariant code that makes decisions based on whether a pointer represents a struct will get confused. The input value represents a struct. The return value represents a pointer to struct so I'm making a copy of the pointer.